### ISSUE TITLE
Raw Queries not working

### ISSUE THREAD

**Issue by prashantsmore** [susanBuck/dwa15-spring2018#74]
I am calling the below method to find matching records. As can be seen , I have a raw statement 
as follows, but when the results are displayed , it shows all the records for employees.
In the debug window, the query which is executed is "select * from employees"
I am expecting the complete raw statement to be executed
```  
 public function match($id)
 {
 $matchEmployees = Employee::raw( 'SELECT * FROM employees where id in 
( SELECT employee_id FROM employee_skill where skill_id =  (select id from skills where name = (select skill_required from staff_requests where id = ?)))' )
 ->setBindings([$id])
 ->get();  
 	return view('staffRequests.match')->with([
            'matchEmployees' => $matchEmployees,
        ]);
    }
```

**Comment by susanBuck** [comment]
The Query Builder and Eloquent are both designed to construct SQL statements.

The purpose of the `raw` method is to allow you to inject chunks of your own SQL into the statements that are constructed. 

It's not designed to just execute a complete SQL statement, so there's a conflict in how you're trying to use it. 

Case in point, part of your statement says `SELECT * FROM employees`, but that's redundant because when you use the `Employee` model it's going to construct a statement to query the `employees` table. Then the `get` method is used to do a SELECT, so that's also redundant.

To demonstrate, I create a scaled down version of what you're trying to execute (just to simplify the example):

```php
 $matches = Book::raw('SELECT * FROM books WHERE id = 1')->get();
```

This produces the same unexpected results you're describing - rather than getting the book with id = 1, it's getting all the books (`SELECT * FROM books`).

The correct approach would look something more like this:
```php
$matches = Book::whereRaw('id = 1')->get();
```

Note that there's no SELECT * FROM books in the raw statement; that's taken care of by using the `Book` model and `get`.

In short, you're fighting against the tools you're working with. :-) Eloquent and/or the Query Builder want to construct the SQL statement for you, but you're trying to circumvent that by feeding them your own complete SQL statements.

(Also, side note: the above technique (raw SQL) is not in sync with the project/curriculum expectations. You're expected to demonstrate use of Eloquent and all the various relationship methods.)

**Comment by prashantsmore** [comment]
Thank you Susan. Inputs are much appreciated.
